[Feat] Feat/75 feat 파이프라인 job 생성

Shin-JiHyun opened the pull request:
**1. 변경 요약 (Summary of Changes)**
- Jenkins 서버로 파이프라인 Job을 생성하는 기능 구현
- Job 생성 요청 시 pipeline DSL script를 포함한 XML payload 전송
- XML 정의 및 URL 매핑 정리 (`/createItem?name={jobName}`)
- swagger 설정 추가(접근 경로 권한 허용 및 group 추가)
- 
![image](https://github.com/user-attachments/assets/bf540f00-2e52-4b71-b1a5-6cf1b056a4d5)


**2. 테스트 방법 (How to Test)**
- Swagger UI를 통해 아래 경로에서 API 테스트 가능:
    - 주소: `http://localhost:8080/api/swagger-ui`
    - 경로: `POST /api/jenkins/job/create/pipeline-script`

- 요청 예시:
-  jenkins info를 먼저 등록합니다
- 
![image](https://github.com/user-attachments/assets/322860ea-6240-4c5d-af53-ec3237e3545e)

- job을 create 합니다.
- 
![image](https://github.com/user-attachments/assets/254c2ad5-93c0-4e72-ac5d-55e309843883)

젠킨스 대시보드에서 생성된 job을 확인합니다.
![image](https://github.com/user-attachments/assets/8b3415e2-c561-47f2-a68e-fc5fff171cc1)

- 예상 응답:
    - 성공 시: HTTP 200 OK, Jenkins에 Job 생성됨
    - 실패 시: 401 (인증 오류), 404 (URI 오류), 500 (서버 내부 오류 등)

- 테스트 시나리오:
    - 단위 테스트: Job 생성 요청 성공 및 실패 케이스 검증
    - 통합 테스트: 실 Jenkins 서버와 연결하여 end-to-end 테스트
    - e2e 테스트: Swagger 및 Postman을 통해 API 연동 정상 동작 확인

**3. 관련 이슈 (Related Issues)**
- 연결된 이슈 번호: `#75` (Jenkins Job 생성 기능)

**4. 체크리스트 (Checklist)**
- [x] 코드 컨벤션을 준수했는가? (파일명, 네이밍, 포맷 등)
- [x] 새로운 기능/버그 수정에 대한 단위 테스트 작성 및 통과했는가?
- [x] 통합 테스트 및 시나리오 테스트를 통과했는가?
- [ ] 문서(Docs) 업데이트가 필요한 경우 반영했는가? (예: API 명세)
- [x] 주요 로직에 적절한 주석이 포함되었는가?
- [x] 보안/성능 고려 사항 검토했는가? (예: Jenkins 인증처리)
- [x] 관련 브랜치 전략 및 머지 대상이 적절한가? (e.g. `dev` → `main`)

Shin-JiHyun committed:
[Feat] 젠킨스 테스트를 위한 swagger 설정
- group에 job-api 추가
- swagger 경로 접근 허용

Shin-JiHyun committed:
[Feat] 젠킨스 job 생성
- Jenkins createItem API 연동
- pipeline DSL script 기반 XML 전송 구조 구성
- item 타입 추가로 freeStyle 경로 변경
- jenkinsInfo에 pipelineScript 매핑 추가

Shin-JiHyun committed:
Merge branch 'refs/heads/develop' into feat/75-feat-파이프라인-job-생성

# Conflicts:
#	backend/src/main/java/com/example/backend/config/SecurityConfig.java
#	backend/src/main/java/com/example/backend/config/jwt/JwtAuthenticationFilter.java

Shin-JiHyun committed:
[Fix] 젠킨스 job 경로 변경

- item 타입 추가로 경로 변경

chan0o0seo commented on backend/src/main/java/com/example/backend/jenkins/job/model/pipeline/PipelineScript.java:
```diff
@@ -0,0 +1,53 @@
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class PipelineScript {
```
직관적으로 Pipeline으로 하는게 나을듯

Shin-JiHyun replied:
젠킨스 파일로 하는건 추후에 추가안하는거지?? 그거 때문에 이렇게 이름 해놨는데

chan0o0seo commented on backend/src/main/java/com/example/backend/jenkins/job/model/pipeline/PipelineScriptHistory.java:
```diff
@@ -0,0 +1,62 @@
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class PipelineScriptHistory {
```
여기도 PipelineHistory

chan0o0seo commented on backend/src/main/java/com/example/backend/jenkins/job/repository/PipelineScriptRepository.java:
```diff
@@ -0,0 +1,7 @@
+import com.example.backend.jenkins.job.model.pipeline.PipelineScript;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface PipelineScriptRepository extends JpaRepository<PipelineScript, Long> {
```
id 타입은 UUID인데 Long 타입을 지정했어요

Shin-JiHyun replied:
헉 저런..

chan0o0seo commented on backend/src/main/java/com/example/backend/config/jwt/JwtAuthenticationFilter.java:
```diff
@@ -29,14 +29,19 @@ protected void doFilterInternal(HttpServletRequest request,
                 path.startsWith("/oauth2/") ||
-                path.startsWith("/login/oauth2/") ||
-                path.startsWith("/api/auth/email") ||
-                path.startsWith("/api/auth/token")
```
token 경로가 없어졌고
 path.startsWith("/swagger-ui"), path.startsWith("/swagger-resources")
 
이거 보다는

path.startsWith("/swagger") 이렇게 하면 다 포함될거같은데

chan0o0seo commented on backend/src/main/java/com/example/backend/config/SecurityConfig.java:
```diff
@@ -39,8 +39,10 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+                                "/api/email/**",
+                                "/swagger-ui/**",
+                                "/v3/api-docs/**",
+                                "/swagger-ui.html"
```
"/swagger**" 이렇게 하면 "/swagger-ui.html", "/swagger-ui/**", 이거 2개 다 커버될거같고, token경로 허용 사라졌어

Shin-JiHyun committed:
[Fix] 코드리뷰 후 수정
- swagger 경로 하나로 합침
- 젠킨스 scm 구현 안하기로 결정함에 따라 클래스 이름pipelineScript-> pipeline으로 수정
- 레포지토리 타입 Long-> UUID로 수정
